app/grandchallenge/components/models.py

```diff
@@ -112,6 +112,8 @@ class InterfaceKindChoices(models.TextChoices):
     IMAGE = "IMG", _("Image")
     SEGMENTATION = "SEG", _("Segmentation")
     HEAT_MAP = "HMAP", _("Heat Map")
+    DISPLACEMENT_FIELD = "DSPF", _("Displacement field")
+    DEFORMATION_FIELD = "DEFF", _("Deformation field")
```

Comment:
What is the difference between the two? They seem to share all of the same validation methods.

Reply:
The values in the vector-field are different, I cannot quite quote the definition that Alessa and people from FME give because I always found it a bit confusing, but I can write it down using math:

Let's assume we want to map a position $\overrightarrow{v_A}$ to $\overrightarrow{v_B}$ using one of the two types of vector fields. This works like this:

- $D$ as a deformation field: $\overrightarrow{v_B} = D\( \overrightarrow{v_A} \)$
- $D$ as a displacement field: $\overrightarrow{v_B} = \overrightarrow{v_A} + D\( \overrightarrow{v_A} \)$

And that is all. So the semantics of the two are different, but the two cases can easily be implemented in one go in CIRRUS, so I added both.

Reply:
If we want different types in GC we need to be able to have some rule that distinguishes between the two, otherwise they should be the same type as far as GC is concerned. What would that rule be?

Reply:
> rule that distinguishes between the two

Not _quite sure_ what you mean here. The difference between a deformation field and a displacement field is the same as an MRI and an CT scan, I think. The data format they use for storing is the same (3d vector fields) but the semantics are different, which I thought we encode in the ComponentInterface. It is like `cardiac-ct` vs `cardiac-mri`, but this time also adding a different _kind_ to allow CIRRUS to lock on to the different semantics in a structured way.

Would you suggest another kind of property that we add to a component interface? That would work, too. Ultimately, a user needs to tag if something is a deformation of displacement field. There is no way for us to know what the thing they upload is - just like with MRI vs CT scans (ignoring the existence of potential related DICOM tags for second here).

Reply:
For example, what is a heat map and what is a segmentation? A Heat map has continuous values, a Segmentation has discrete values. When we get something new we need to be able to distinguish between the two, and we do that on validation. If we have two new types we need to be able to distinguish between them somehow. Otherwise it should be one type.

Reply:
Comes down to - if a user provides a DISPLACEMENT_FIELD when a DEFORMATION_FIELD is expected it should be rejected at import time, not at CIRRUS run time.

Reply:
> Comes down to - if a user provides a DISPLACEMENT_FIELD when a DEFORMATION_FIELD is expected it should be rejected at import time, not at CIRRUS run time.

Ok, I see. Problem with this is that the two _types_ of vector fields are _fully compatible_. A displacement field looks _exactly_ like a deformation field. Only the offsets are different leading to an incorrect interpretation when attempting to use a deformation field as a displacement field and vice versa. CIRRUS cannot detect this either when shown, the only thing that would happen is that the mapped coordinates are all wrong.

To decide if a displacement field is "correct" one needs to evaluate if it actually "does the coordinate mapping correctly" which is something we cannot compute. (I would not even algorithmically know how to do this in a simple and reliable way...)

Also, right now it is possible to upload a heat map as a mri image, right? And one could upload a displacement field as a mri image which then would be interpreted as a 3d MRI image with multiple time points (namely 3). Therefore, I would argue that we already make semantic decisions based on the interface-kind. I can see that displacement and deformation fields are more closely related than, for example, an 4d image vs displacement fields and it might be sub-optimal to distinguish between these types on the interface-kind level. But we need the user to tell us what they actually uploaded.

From the little maths-section I show above, you can derive that deformation fields are equivalent to displacement fields as long as image $A$ is available. Having image $A$ allows one to add or remove the offset $\overrightarrow{v_A}$ from $D$ to convert one into the other type. In CIRRUS this can be done because we have image $A$. But also here the user has to tell us in which direction this conversion has to be applied to make sense.

Only solutions I see here are:
- Drop support for deformation fields to avoid this confusion
    - Alessa told me that this mostly would work, but I know that it would be somewhat unfortunate because it requires existing registrations to be recomputed. Sort of a bike-shedding thing that we would do here. There are pretty much no complication of _applying_ the conversion in CIRRUS so this would be an unfortunate miss, imo.
- Collect the deformation/displacement semantics in some other way during the upload process - happy to hear suggestions here. As I outlined, there is unfortunately no way for us to automatically latch on to the type of vector field that is uploaded because they use the exact same format.
- (Keep this implementation... of course this is an option, too)

Reply:
> Drop support for deformation fields to avoid this confusion

Do this, that way it fits in with the existing design of the application.

Reply:
Done!